### PR TITLE
Update extensions to build 2017-06-13

### PR DESCRIPTION
**Pull request by ewpatton**
This PR includes updated versions of the extensions to address a number of bugs reported on the forum as well as to add a few new blocks to aid in getting information about the connected device and its services in a more App Inventor friendly way.

### CODE REVIEW

**Comment by ewpatton** [comment]
Do not merge yet. I am looking into two more possible bugs and will update this commit.

**Comment by ewpatton** [comment]
This is now the latest version of the extensions. It includes some additional bug fixes and adds the AutoReconnect and ConnectionTimeout properties to the BluetoothLE extension.

**Comment by ewpatton** [comment]
Do not merge (again). Some extensions need to have URLs fixed based on issue #134

**Comment by ewpatton** [comment]
Fixed links to PDFs in accel and gyro components of Arduino 101. Removed links for Arduino Pins since the documents do not exist.

**Comment by ewpatton** [comment]
This will need rebasing/updating after #137 is merged. Do not merge.

**Comment by ewpatton** [comment]
This should be the final version of extensions and documentation barring any requests for changes during review. @marksherman @afmckinney

File: docs/src/app/pages/MicroBitMagnetometer.md [hunk 1]
```diff
@@ -1,10 +1,6 @@
-# Microbit_Magnetometer
+# Microbit\_Magnetometer
 
-The <code>Microbit_Magnetometer</code> provides App Inventor users with the ability toconfigure the BBC micro:bit's on-board magnetometer and receive one or more magnetometer samples via the appropriate methods.<br>
-
-The magnetometer's report rate, or period, determines how frequently data will be sent to App Inventor.<br>
-
-<strong>More links:</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/MicrobitMagnetometer.aia' target='_blank'>sample project</a> for the BBC micro:bit Magnetometer.</li><li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_IoT_Microbit_Magnetometer.pdf' target='_blank'>how to instructions</a> for the micro:bit Magnetometer.</li></ul>
+The <code>Microbit_Magnetometer</code> provides App Inventor users with the ability toconfigure the BBC micro:bit's on-board magnetometer and receive one or more magnetometer samples via the appropriate methods.
```
Comment (marksherman) [inline docs/src/app/pages/MicroBitMagnetometer.md]: "toconfigure"

File: docs/src/app/pages/MicroBitUart.md [hunk 2]
```diff
@@ -14,25 +14,52 @@ Component for Microbit_Uart
 
 ## Methods
 
-+ <a name="RequestTXCharacteristic"></a>`RequestTXCharacteristic` – Method for RequestTXCharacteristic
++ <a name="RequestTXCharacteristic"></a>`RequestTXCharacteristic` – Request notifications of messages received from the micro:bit's serial UART. New messages will
```
Comment (marksherman) [inline docs/src/app/pages/MicroBitUart.md]: Request -> Requests

File: docs/src/app/pages/MicroBitUart.md [hunk 3]
```diff
@@ -14,25 +14,52 @@ Component for Microbit_Uart
...
+ be received through the
+ <a href='#TXCharacteristicReceived'><code>TXCharacteristicReceived</code></a>
+ event.
 
 ![call Microbit_Uart1 RequestTXCharacteristic](blocks/Microbit_Uart.RequestTXCharacteristic.svg)
 
-+ <a name="StopTXCharacteristicUpdates"></a>`StopTXCharacteristicUpdates` – Method for StopTXCharacteristicUpdates
++ <a name="StopTXCharacteristicUpdates"></a>`StopTXCharacteristicUpdates` – Stop receiving updates from the micro:bit's serial UART. Note that there may be pending
```
Comment (marksherman) [inline docs/src/app/pages/MicroBitUart.md]: Stop -> Stops

File: docs/src/app/pages/Arduino101Pins.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 # Arduino101Pins
 
-The Arduino101Pins extension provides a general purpose interface to any hardware connected to the Arduino's digital or analog I/O pins.<br>
+The Arduino101Pins extension provides a general purpose interface to any hardware connected to the Arduino's digital or analog I/O pins.<!--<br>
 
-<strong>More links</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/Arduino101Pins.aia' target='_blank'>sample project</a> for the Arduino 101 Pins.</li><li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_IoT_Pins.pdf' target='_blank'>how to instructions</a> for the Arduino 101 Pins extension.</li></ul>
+<strong>More links</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/Arduino101Pins.aia' target='_blank'>sample project</a> for the Arduino 101 Pins.</li><li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_IoT_Pins.pdf' target='_blank'>how to instructions</a> for the Arduino 101 Pins extension.</li></ul>-->
```
Comment (marksherman) [inline docs/src/app/pages/Arduino101Pins.md]: `-->` is rendering, along with the whole paragraph that you tried to comment out.

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 5]
```diff
@@ -19,21 +19,52 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+
+
+![set BluetoothLE1 AutoReconnect  to](blocks/BluetoothLE.AutoReconnect_setter.svg)
+
 + <a name="BatteryValue"></a>`BatteryValue` – Returns the battery level.
 
 
 ![get BluetoothLE1 BatteryValue ](blocks/BluetoothLE.BatteryValue_getter.svg)
 
++ <a name="ConnectedDeviceName"></a>`ConnectedDeviceName` –
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: no descriptive sentence?

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 6]
```diff
@@ -19,21 +19,52 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+
+
+![get BluetoothLE1 ConnectedDeviceName ](blocks/BluetoothLE.ConnectedDeviceName_getter.svg)
+
 + <a name="ConnectedDeviceRssi"></a>`ConnectedDeviceRssi` – Returns the RSSI (Received Signal Strength Indicator) of connected device.
 
 
 ![get BluetoothLE1 ConnectedDeviceRssi ](blocks/BluetoothLE.ConnectedDeviceRssi_getter.svg)
 
++ <a name="ConnectionTimeout"></a>`ConnectionTimeout` – The amount of time, in seconds, that the BluetoothLE component will wait for a connection to be established with a device when <a href='#Connect'></code>Connection</code></a> or <a href='#ConnectWithAddress'><code>ConnectWithAddress</code></a>. If a connection is not established in the given amount of time, the attempt will be aborted and the <a href='#ConnectionFailed'><code>ConnectionFailed</code></a> event will be run.
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: > The amount of time, in seconds, that the BluetoothLE component will wait for a connection to be established with a device when Connection or ConnectWithAddress.

when what?

Also, "Connection" and "ConnectWithAddress" are rendering differently. The first code tag is a close-tag.

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 7]
```diff
@@ -19,21 +19,52 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+
+
+![get BluetoothLE1 DeviceCharacteristics ](blocks/BluetoothLE.DeviceCharacteristics_getter.svg)
+
 + <a name="DeviceList"></a>`DeviceList` – Returns a sorted list of BluetoothLE devices as a String.
 
 
 ![get BluetoothLE1 DeviceList ](blocks/BluetoothLE.DeviceList_getter.svg)
 
++ <a name="DeviceServices"></a>`DeviceServices` –
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: no descriptive sentence

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 8]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+       The unique identifier of the service passed in the read or register call.
+     * <code>characteristicUuid</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/text.html#string">_text_</a>) &mdash;
+       The unique identifier of the characteristic in the read or register call.
+     * <code>signed</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/logic.html#true" target="_blank">_boolean_</a>)
+       Interpret the shorts as signed (true) or unsigned (false).
 
-+ <a name="ReadStrings"></a>`ReadStrings` – Reads one or more strings from a connected Bluetooth device.
+![call BluetoothLE1 ReadShortsserviceUuidcharacteristicUuidsigned](blocks/BluetoothLE.ReadShorts.svg)
+
++ <a name="ReadStrings"></a>`ReadStrings` – Reads one or more null-terminated strings from a connected BluetoothLE device. Service Unique ID
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: "Unique ID" and "UniqueID" both present in same sentence. Based on other entries the space should be used.

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 9]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+       The unique identifier of the service passed in the read or register call.
+     * <code>characteristicUuid</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/text.html#string">_text_</a>) &mdash;
+       The unique identifier of the characteristic in the read or register call.
+     * <code>signed</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/logic.html#true" target="_blank">_boolean_</a>)
+       Interpret the bytes as signed (true) or unsigned (false).
 
 ![call BluetoothLE1 RegisterForBytesserviceUuidcharacteristicUuidsigned](blocks/BluetoothLE.RegisterForBytes.svg)
 
-+ <a name="RegisterForFloats"></a>`RegisterForFloats` – Registers to receive updates when one or more floating point values from a connected Bluetooth device change.
++ <a name="RegisterForFloats"></a>`RegisterForFloats` – Register to receive updates when one or more floating point values from a connected Bluetooth device change.
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: should indeed be "Registers"

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 10]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+       Interpret the bytes as signed (true) or unsigned (false).
+     * <code>values</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/lists.html#makealist">_list_</a>) &mdash;
+       A list of values to write to the device.
 
 ![call BluetoothLE1 WriteBytesWithResponseserviceUuidcharacteristicUuidsignedvalues](blocks/BluetoothLE.WriteBytesWithResponse.svg)
 
-+ <a name="WriteFloats"></a>`WriteFloats` – Writes one or more floating point values to a connected BluetoothLE device. Service Unique ID and Characteristic Unique ID are required. The values parameter can either be a single numeric value or a list of values. If shortFloat is true, the values will be compressed to 16-bits. If shortFloat is false, the values are sent as 32-bit.
++ <a name="WriteFloats"></a>`WriteFloats` – Writes one or more IEEE 754 floating point numbers to a connected BluetoothLE device. Service Unique ID
+ and Characteristic Unique ID are required. The values parameter can either be a single numeric
+ value or a list of values. If <code>shortFloats</code> is true, then each numeric value will be
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: Block image says "shortFloat" without the 's'. The description says "shortFloats" three times

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 11]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+       A list of values to write to the device.
 
 ![call BluetoothLE1 WriteFloatsserviceUuidcharacteristicUuidshortFloatvalues](blocks/BluetoothLE.WriteFloats.svg)
 
-+ <a name="WriteFloatsWithResponse"></a>`WriteFloatsWithResponse` – Writes one or more 8-bit integer values to a connected BluetoothLE device and wait for an acknowledgement via the FloatValuesWritten event. Service Unique ID and Characteristic Unique ID are required. The values parameter can either be a single numeric value or a list of values. If shortFloat is true, the values will be compressed to 16-bits. If shortFloat is false, the values are sent as 32-bit.
++ <a name="WriteFloatsWithResponse"></a>`WriteFloatsWithResponse` – Writes one or more IEEE 754 floating point values to a connected BluetoothLE device and waits for an
+ acknowledgement via the <a href="#FloatsWritten"><code>FloatsWritten</code></a> event.
+ Service Unique ID and Characteristic Unique ID are required. The values parameter can either
+ be a single numeric value or a list of values. If <code>shortFloats</code> is false,
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: same thing: "shortFloat" in the block image but "shortFloats" in the text

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 12]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+     * <code>values</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/lists.html#makealist">_list_</a>) &mdash;
+       A list of values to write to the device.
 
 ![call BluetoothLE1 WriteFloatsWithResponseserviceUuidcharacteristicUuidshortFloatvalues](blocks/BluetoothLE.WriteFloatsWithResponse.svg)
 
-+ <a name="WriteIntegers"></a>`WriteIntegers` – Writes one or more 32-bit integer values to a connected BluetoothLE device. Service Unique ID and Characteristic Unique ID are required. The values parameter can either be a single numeric value or a list of values. If signed is true, the acceptable values are between -2147483648 and 2147483647. If signed is false, the acceptable values are between 0 and 4294967295.
++ <a name="WriteIntegers"></a>`WriteIntegers` – Writes one or more 32-bit integer values to a connected BluetoothLE device. Service Unique ID
+ and Characteristic Unique ID are required. The values parameter can either be a single numeric
+ value or a list of values. If signed is true, the acceptable values are between -2147483648 and
+ 2147483647. If signed is false, the acceptable values are between 0 and 4294967295.
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: rendering as the beginning of a numbered list. Remove line break and it should be ok, I hope.

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 13]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+       Interpret the integers as signed (true) or unsigned (false).
+     * <code>values</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/lists.html#makealist">_list_</a>) &mdash;
+       A list of values to write to the device.
 
 ![call BluetoothLE1 WriteIntegersWithResponseserviceUuidcharacteristicUuidsignedvalues](blocks/BluetoothLE.WriteIntegersWithResponse.svg)
 
-+ <a name="WriteShorts"></a>`WriteShorts` – Writes one or more 16-bit integer values to a connected BluetoothLE device. Service Unique ID and Characteristic Unique ID are required. The values parameter can either be a single numeric value or a list of values. If signed is true, the acceptable values are between -32768 and 32767. If signed is false, the acceptable values are between 0 and 65535.
++ <a name="WriteShorts"></a>`WriteShorts` – Writes one or more 16-bit integer values to a connected BluetoothLE device. Service Unique ID
+ and Characteristic Unique ID are required. The values parameter can either be a single numeric
+ value or a list of values. If signed is true, the acceptable values are between -32768 and 32767.
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: codeify "signed"

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 14]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+       A list of values to write to the device.
 
 ![call BluetoothLE1 WriteIntegersserviceUuidcharacteristicUuidsignedvalues](blocks/BluetoothLE.WriteIntegers.svg)
 
-+ <a name="WriteIntegersWithResponse"></a>`WriteIntegersWithResponse` – Writes one or more 32-bit integer values to a connected BluetoothLE device and wait for an acknowledgement via the IntegerValuesWritten event. Service Unique ID and Characteristic Unique ID are required. The values parameter can either be a single numeric value or a list of values. If signed is true, the acceptable values are between -2147483648 and 2147483647. If signed is false, the acceptable values are between 0 and 4294967295.
++ <a name="WriteIntegersWithResponse"></a>`WriteIntegersWithResponse` – Writes one or more 32-bit integer values to a connected BluetoothLE device and waits for an
+ acknowledgement via the <a href="#IntegersWritten"><code>IntegersWritten</code></a> event.
+ Service Unique ID and Characteristic Unique ID are required. The values parameter can either
+ be a single numeric value or a list of values. If signed is true, the acceptable values are
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: codeify "signed"

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 15]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+       A list of values to write to the device.
 
 ![call BluetoothLE1 WriteShortsserviceUuidcharacteristicUuidsignedvalues](blocks/BluetoothLE.WriteShorts.svg)
 
-+ <a name="WriteShortsWithResponse"></a>`WriteShortsWithResponse` – Writes one or more 16-bit integer values to a connected BluetoothLE device and wait for an acknowledgement via the ShortValuesWritten event. Service Unique ID and Characteristic Unique ID are required. The values parameter can either be a single numeric value or a list of values. If signed is true, the acceptable values are between -32768 and 32767. If signed is false, the acceptable values are between 0 and 65535.
++ <a name="WriteShortsWithResponse"></a>`WriteShortsWithResponse` – Writes one or more 16-bit integer values to a connected BluetoothLE device and waits for an
+ acknowledgement via the <a href="#ShortsWritten"><code>ShortsWritten</code></a> event.
+ Service Unique ID and Characteristic Unique ID are required. The values parameter can either
+ be a single numeric value or a list of values. If signed is true, the acceptable values are
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: codeify "signed"

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 16]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+ encoding.
+
+ __Parameters__:
+
+     * <code>serviceUuid</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/text.html#string">_text_</a>) &mdash;
+       The unique identifier of the service passed in the read or register call.
+     * <code>characteristicUuid</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/text.html#string">_text_</a>) &mdash;
+       The unique identifier of the characteristic in the read or register call.
+     * <code>utf16</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/logic.html#true" target="_blank">_boolean_</a>)
+       Send the string encoded as UTF-8 (false) or UTF-16 (true) code points.
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: say true-case first (this bug may exist elsewhere, too)

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 17]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
+ encoding.
+
+ __Parameters__:
+
+     * <code>serviceUuid</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/text.html#string">_text_</a>) &mdash;
+       The unique identifier of the service passed in the read or register call.
+     * <code>characteristicUuid</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/text.html#string">_text_</a>) &mdash;
+       The unique identifier of the characteristic in the read or register call.
+     * <code>utf16</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/logic.html#true" target="_blank">_boolean_</a>)
+       Send the string encoded as UTF-8 (false) or UTF-16 little endian (true) code points.
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: definitely exists elsewhere. say true case first

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 18]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
++ <a name="Disconnected"></a>`Disconnected` – The <code>Disconnected</code> event is run when a Bluetooth low energy device is disconnected.
+ This can be caused by a call to <a href="#Disconnect"><code>Disconnect</code></a> or
+ <a href="#DisconnectWithAddress"><code>DisconnectWithAddress</code></a>, or after a device
+ is moved away or reset such that a loss of connection occurs.
 
 ![when BluetoothLE1 Disconnecteddo](blocks/BluetoothLE.Disconnected.svg)
 
-+ <a name="FloatsReceived"></a>`FloatsReceived` – Triggers event when one or more floating point values from a BluetoothLE device are received.
++ <a name="FloatsReceived"></a>`FloatsReceived` – The <code>FloatsReceived</code> event is run when one or more IEEE 754 floating point values are received from a
+ connected Bluetooth device. Depending on the <code>shortFloats</code> parameter of the last call to
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: is it shortFloats or shortFloat (with or without suffix s)

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 19]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
 
 ![when BluetoothLE1 FloatsWritten serviceUuid characteristicUuid floatValues do](blocks/BluetoothLE.FloatsWritten.svg)
 
-+ <a name="IntegersReceived"></a>`IntegersReceived` – Triggers event when one or more integer values from a connected BluetoothLE device are received.
++ <a name="IntegersReceived"></a>`IntegersReceived` – The <code>IntegersReceived</code> event is run when one or more 32-bit integer values are received from a
+ connected Bluetooth device. Depending on the <code>sign</code> parameter of the last call to
+ <a href="#ReadIntegers"><code>ReadIntegers</code></a> or
+ <a href="#RegisterForIntegers"><code>RegisterForIntegers</code></a> for the given
+ <code>serviceUuid</code> and <code>characteristicUuid</code>, the <code>intValues</code> list
+ will contain numbers ranging from 0 to 4294967296 (<code>signed = false</code>) or
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: sign or signed?

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 20]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
-![when BluetoothLE1 RssiChanged device_rssi do](blocks/BluetoothLE.RssiChanged.svg)
+![when BluetoothLE1 RssiChanged rssi do](blocks/BluetoothLE.RssiChanged.svg)
 
-+ <a name="ShortsReceived"></a>`ShortsReceived` – Triggers event when one or more short values from a connected BluetoothLE device are received.
++ <a name="ShortsReceived"></a>`ShortsReceived` – The <code>ShortsReceived</code> event is run when one or more short integer values are received from a
+ connected Bluetooth device. Depending on the <code>sign</code> parameter of the last call to
+ <a href="#ReadShorts"><code>ReadShorts</code></a> or
+ <a href="#RegisterForShorts"><code>RegisterForShorts</code></a> for the given
+ <code>serviceUuid</code> and <code>characteristicUuid</code>, the <code>shortValues</code> list
+ will contain numbers ranging from 0 to 65535 (<code>signed = false</code>) or -32768 to 32767
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: sign or signed?

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 21]
```diff
@@ -53,226 +89,740 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new p
...
 
 ![when BluetoothLE1 ShortsWritten serviceUuid characteristicUuid shortValues do](blocks/BluetoothLE.ShortsWritten.svg)
 
-+ <a name="StringsReceived"></a>`StringsReceived` – Triggers event when one or more string values from a connected BluetoothLE device a received.
++ <a name="StringsReceived"></a>`StringsReceived` – The <code>StringsReceived</code> event is run when one or more strings are received from a
+ connected Bluetooth device. Depending on the <code>utf16</code> parameter of the last call to
+ <a href="#ReadStrings"><code>ReadStrings</code></a> or
+ <a href="#RegisterForStrings"><code>RegisterForStrings</code></a> for the given
+ <code>serviceUuid</code> and <code>characteristicUuid</code>, the <code>stringValues</code> list
+ will contain either a UTF-8 decoded (<code>utf16 = false</code>) or UTF-16 little endian
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: say true first

File: docs/src/app/pages/MicroBitIoPin.md [hunk 22]
```diff
@@ -1,8 +1,6 @@
-# Microbit_Io_Pin
+# Microbit\_Io\_Pin
 
-The <code>Microbit_Io_Pin</code> component lets users configure the BBC micro:bit's analog pins for input and output, and to read, write, and request notifications for theI/O pin states.<br>
-
-<strong>More links:</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/MicrobitIoPins.aia' target='_blank'>sample project</> for the micro:bit IO pins.</li><li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_IoT_Microbit_IO_Pins.pdf' target='_blank'>how to instructions</a> for the micro:bit IO pins.</li></ul>
+The <code>Microbit_Io_Pin</code> component lets users configure the BBC micro:bit's analog pins for input and output, and to read, write, and request notifications for theI/O pin states.
```
Comment (marksherman) [inline docs/src/app/pages/MicroBitIoPin.md]: space between theI/O

File: docs/src/app/pages/MicroBitUart.md [hunk 23]
```diff
@@ -1,10 +1,10 @@
-# Microbit_Uart
+# Microbit\_Uart
 
-Component for Microbit_Uart
+The <code>Microbit_Uart</code> sensor provides App Inventor users with the ability to read from and write strings to the BBC micro:bit's serial UART port.
```
Comment (marksherman) [inline docs/src/app/pages/MicroBitUart.md]: why mention users? Just say "provides" 
(Also, Hal came in today with a new mission that will be against the word "users")

File: docs/src/app/pages/MicroBitUart.md [hunk 24]
```diff
@@ -14,25 +14,52 @@ Component for Microbit_Uart
...
 
-+ <a name="StopTXCharacteristicUpdates"></a>`StopTXCharacteristicUpdates` – Method for StopTXCharacteristicUpdates
++ <a name="StopTXCharacteristicUpdates"></a>`StopTXCharacteristicUpdates` – Stop receiving updates from the micro:bit's serial UART. Note that there may be pending
+ messages from the device that will still be reported through the
+ <a href='#TXCharacteristicReceived'><code>TXCharacteristicReceived</code></a> event.
 
 ![call Microbit_Uart1 StopTXCharacteristicUpdates](blocks/Microbit_Uart.StopTXCharacteristicUpdates.svg)
 
-+ <a name="WriteRXCharacteristic"></a>`WriteRXCharacteristic` – Method for WriteRXCharacteristic
++ <a name="WriteRXCharacteristic"></a>`WriteRXCharacteristic` – Write the given value <code>UART_TX</code> to the micro:bit. The size of the message will
```
Comment (marksherman) [inline docs/src/app/pages/MicroBitUart.md]: Writes

**Review by marksherman — CHANGES_REQUESTED** [review]

**Comment by ewpatton** [comment]
I've pushed updated documentation that should address these issues (and a few that weren't documented but of the same variety).

File: docs/src/app/pages/BluetoothLEIntro.md [hunk 25]
```diff
@@ -282,7 +282,7 @@ Bluetooth Low Energy, also referred to as Bluetooth LE or simply BLE, is a new c
      * <code>characteristicUuid</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/text.html#string">_text_</a>) &mdash;
        The unique identifier of the characteristic in the read or register call.
      * <code>utf16</code> (<a href="http://appinventor.mit.edu/explore/ai2/support/blocks/logic.html#true" target="_blank">_boolean_</a>)
-       Interpret the string content as UTF-8 (false) or UTF-16 (true) code points.
+       Interpret the string content as UTF-16 (true) or UTF-9 (false) code points.
```
Comment (marksherman) [inline docs/src/app/pages/BluetoothLEIntro.md]: UTF-9?

**Review by marksherman — CHANGES_REQUESTED** [review]
I love all your new intro wordings. Just one typo.

**Pull request merged**